src/ui/components/UploadScreen/SettingsPreview.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Workspace } from "ui/types";
+
+const getIconAndText = (
+  isPublic: boolean,
+  selectedWorkspaceId: string | null,
+  workspaceName: string
+) => {
+  let text, icon;
+
+  if (!isPublic) {
+    if (selectedWorkspaceId) {
+      text = `Shared privately with ${workspaceName}`;
+      icon = "groups";
+    } else {
+      text = `Only you can view this`;
+      icon = "person";
+    }
+  } else {
+    text = "This replay can be viewed by anyone with the link";
+    icon = "public";
+  }
+
+  return { text, icon };
+};
+
+export default function SettingsPreview({
+  onClick,
+  isPublic,
+  workspaces,
+  selectedWorkspaceId,
+}: {
+  onClick: () => void;
+  isPublic: boolean;
+  workspaces: Workspace[];
+  selectedWorkspaceId: string | null;
```

Comment:
Is this enough props that it feels worth pulling out into a separate type just to clean up this signature?